warehouse/manage/views.py

```diff
@@ -1367,6 +1378,132 @@ def delete_organization(self):
+    context=Organization,
+    renderer="manage/organization/projects.html",
+    uses_session=True,
+    require_csrf=True,
+    require_methods=False,
+    permission="view:organization",
+    has_translations=True,
+    require_reauth=True,
+)
+class ManageOrganizationProjectsViews:
+    def __init__(self, organization, request):
+        self.organization = organization
+        self.request = request
+        self.user_service = request.find_service(IUserService, context=None)
+        self.organization_service = request.find_service(
+            IOrganizationService, context=None
+        )
+        self.project_factory = ProjectFactory(request)
+
+    @property
+    def active_projects(self):
+        return self.organization.projects
+
+    @property
+    def default_response(self):
+        active_projects = self.active_projects
+        all_user_projects = user_projects(self.request)
+        projects_owned = set(
+            project.name for project in all_user_projects["projects_owned"]
+        )
+        projects_sole_owned = set(
+            project.name for project in all_user_projects["projects_sole_owned"]
+        )
+        projects_requiring_2fa = set(
+            project.name for project in all_user_projects["projects_requiring_2fa"]
+        )
+        project_choices = set(
+            project.name
+            for project in all_user_projects["projects_owned"]
+            if not project.organizations
+        )
+        project_factory = self.project_factory
+
+        return {
+            "organization": self.organization,
+            "active_projects": active_projects,
+            "projects_owned": projects_owned,
+            "projects_sole_owned": projects_sole_owned,
+            "projects_requiring_2fa": projects_requiring_2fa,
+            "add_organization_project_form": AddOrganizationProjectForm(
+                self.request.POST,
+                project_choices=project_choices,
+                project_factory=project_factory,
+            ),
+        }
+
+    @view_config(request_method="GET")
+    def manage_organization_projects(self):
+        if self.request.flags.enabled(AdminFlagValue.DISABLE_ORGANIZATIONS):
+            raise HTTPNotFound
+
+        return self.default_response
+
+    @view_config(request_method="POST")
+    def add_organization_project(self):
+        if self.request.flags.enabled(AdminFlagValue.DISABLE_ORGANIZATIONS):
+            raise HTTPNotFound
+
+        # Get and validate form from default response.
+        default_response = self.default_response
+        form = default_response["add_organization_project_form"]
+        if not form.validate():
+            return default_response
+
+        # Get existing project or add new project.
+        if form.add_existing_project.data:
+            project = self.project_factory[form.existing_project_name.data]
+        else:
+            try:
+                project = add_project(form.new_project_name.data, self.request)
```

Comment:
After this line has successfully executed, we will have a new `Project` object and a new `Role` object associating `self.request.user` with the `Project` as `"Owner"`. I believe this is incorrect behavior for the purposes of this feature. The resulting `Role` is independent of membership in `self.organization`, so by leaving or being removed from an organization that user would retain access (not good). `Project` objects associated via `OrganizationProject` objects to an `Organization` should allow organization owners all necessary permissions to manage the resulting `Project`. Since `add_organization_project` should only be available to organization owners the end result _without_ the `Role` is then the same.

Reply:
If a project is owned by an organization but has no individual owners, what happens if the project is removed from the organization? Ways of handling this include:

1. The organization owner who removes the project is added as an individual owner.
2. All organization owners are added as individual owners.
3. We don’t allow this operation. There must be at least one individual owner before the project can be removed from the organization.

Reply:
The "removed" from an organization operation should only be able to happen via deletion or transfer.

Deletion, easy! We remove the OrganizationProject relation, all Releases, and the Project itself.

For Transfer "out" which I now realize we don't have scoped... Yes the third option makes the most sense to me. The Project should have to be transferred to _another_ organization or have at least one Role object with role "Owner" associated with an individual User.